integration_tests/sdk/utils.py

```diff
@@ -199,28 +204,81 @@ def stop_condition(
     )
 
 
+def extract(
+    integration,
+    obj_identifier: Union[DataObject, str],
+    op_name: Optional[str] = None,
+    lazy: bool = False,
+) -> BaseArtifact:
+    """Reads the specified object in from the integration and returns it as an artifact.
+
+    Assumption: the object is a pandas dataframe, serialized in a particular fashion in each integration.
+    This serialization method should match what is done in `save()`.
+    """
+    if isinstance(obj_identifier, DataObject):
+        obj_identifier = obj_identifier.value
+
+    assert isinstance(obj_identifier, str)
+    if isinstance(integration, RelationalDBIntegration):
+        return integration.sql(query="SELECT * from %s" % obj_identifier, name=op_name, lazy=lazy)
+    elif isinstance(integration, S3Integration):
+        return integration.file(
+            obj_identifier, ArtifactType.TABLE, "parquet", name=op_name, lazy=lazy
+        )
+    raise Exception("Unexpected data integration type provided in test: %s", type(integration))
+
+
+def get_object_identifier_from_load_spec(load_spec: LoadSpec) -> str:
```

Comment:
Is this method available for all types of `LoadSpec`s? Could we just add an `.object_identifier()` method to `LoadSpec` object?

Reply:
I like this idea. Yeah, I think every data integration must have some location identifier for the object.